Index: Src/library/vision2/implementation/implementation_interface/widgets/ev_widget_i.e
===================================================================
--- Src/library/vision2/implementation/implementation_interface/widgets/ev_widget_i.e	(revision 94987)
+++ Src/library/vision2/implementation/implementation_interface/widgets/ev_widget_i.e	(working copy)
@@ -102,6 +102,20 @@
 			-- If agent returns `True' then default key processing continues as normal, False prevents
 			-- default key processing from occurring.
 
+	layout: EV_LAYOUTABLE
+			-- The property that defines the widget's attributes in a
+			-- constraint-based layout manager.
+		do
+			Result := internal_layout
+			if Result = Void then
+				create Result
+				internal_layout := Result
+			end
+		ensure
+			result_not_void: Result /= Void
+			internal_set_and_equal_result: internal_layout = Result
+		end
+
 feature -- Status Report
 
 	is_show_requested: BOOLEAN
@@ -309,6 +323,8 @@
 			end
 		end
 
+	internal_layout: EV_LAYOUTABLE
+
 invariant
 	is_displayed_implies_show_requested:
 		is_usable and is_displayed implies is_show_requested
@@ -319,14 +335,14 @@
 		--|is_usable and parent /= Void implies parent.has (interface)
 
 note
-	copyright:	"Copyright (c) 1984-2006, Eiffel Software and others"
+	copyright:	"Copyright (c) 1984-2014, Eiffel Software and others"
 	license:	"Eiffel Forum License v2 (see http://www.eiffel.com/licensing/forum.txt)"
 	source: "[
-			 Eiffel Software
-			 356 Storke Road, Goleta, CA 93117 USA
-			 Telephone 805-685-1006, Fax 805-685-6869
-			 Website http://www.eiffel.com
-			 Customer support http://support.eiffel.com
+			Eiffel Software
+			5949 Hollister Ave., Goleta, CA 93117 USA
+			Telephone 805-685-1006, Fax 805-685-6869
+			Website http://www.eiffel.com
+			Customer support http://support.eiffel.com
 		]"
 
 
Index: Src/library/vision2/interface/widgets/ev_widget.e
===================================================================
--- Src/library/vision2/interface/widgets/ev_widget.e	(revision 94987)
+++ Src/library/vision2/interface/widgets/ev_widget.e	(working copy)
@@ -128,6 +128,13 @@
 			bridge_ok: Result = implementation.default_key_processing_handler
 		end
 
+	layout: EV_LAYOUTABLE
+			-- The property that defines the widget's attributes in a
+			-- constraint-based layout manager.
+		do
+			Result := implementation.layout
+		end
+
 feature {EV_BUILDER} -- Access
 
 	minimum_width_set_by_user: BOOLEAN
@@ -403,7 +410,7 @@
 		is_usable and then attached parent as l_parent implies l_parent.has (Current)
 
 note
-	copyright:	"Copyright (c) 1984-2012, Eiffel Software and others"
+	copyright:	"Copyright (c) 1984-2014, Eiffel Software and others"
 	license:	"Eiffel Forum License v2 (see http://www.eiffel.com/licensing/forum.txt)"
 	source: "[
 			Eiffel Software
Index: Src/library/vision2/vision2.ecf
===================================================================
--- Src/library/vision2/vision2.ecf	(revision 94987)
+++ Src/library/vision2/vision2.ecf	(working copy)
@@ -154,6 +154,7 @@
 				<custom name="vision_implementation" excluded_value="gtk34"/>
 			</condition>
 		</library>
+		<library name="lp_solve" location="$ISE_LIBRARY\contrib\library\math\lp_solve\lp_solve.ecf" />
 		<library name="thread" location="$ISE_LIBRARY\library\thread\thread.ecf"/>
 		<library name="wel" location="$ISE_LIBRARY\library\wel\wel.ecf">
 			<condition>
